modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Utils.java

```diff
@@ -118,4 +118,18 @@ public static void closeChannels(final Collection<Channel> channels) throws IOEx
         }
     }
 
+    public static void maybeDie(Throwable cause) {
+        if (cause instanceof Error) {
+            /*
+             * Here be dragons. We want to rethrow this so that it bubbles up to the uncaught exception handler. Yet, Netty wraps too many
+             * invocations of user-code in try/catch blocks that swallow all throwables. This means that a rethrow here will not bubble up
+             * to where we want it to. So, we fork a thread and throw the exception from there where Netty can not get to it. We do not wrap
+             * the exception so as to not lose the original cause during exit, so we give the thread a name based on the previous stack
+             * frame so that at least we know where it came from.
+             */
+            final StackTraceElement previous = Thread.currentThread().getStackTrace()[2];
+            new Thread(() -> { throw (Error)cause; }, previous.getClassName() + "#" + previous.getMethodName()).start();
```

Comment:
maybe we can try to log the stacktrace in a try-finally block in hopes that we can get the full stacktrace? In the finally we can throw the Error